### ISSUE TITLE
Regression: Invalid docker image layers produced with 0.17.1 (not 0.16.0)

### ISSUE THREAD

**Issue by afirth** [GoogleContainerTools/kaniko#1028]
**Actual behavior**
Running kaniko in cloudbuild on a multistage node build, receive an error extracting a layer during `docker pull`. All ok on 0.16.0.

Error:
```bash
USER@cloudshell:~ (PROJ)$ docker -D pull gcr.io/PROJ/accounts-web-prod:d5b8f8f                                                                                                                                              
d5b8f8f: Pulling from PROJ/accounts-web-prod
3192219afd04: Pull complete
[...]
a792bea52eb7: Pull complete
14b22a746997: Extracting [==================================================>]     503B/503B
ee5fcfa2a3a0: Download complete
ecafe0bd277c: Download complete
aad787893c41: Download complete
failed to register layer: Error processing tar file(exit status 1): no such file or directory
```

cloudbuild step (working at 16)
```yaml
  - name: gcr.io/kaniko-project/executor:v0.16.0
    id: image-build-prod
    waitFor: ["key-docker-copy"]
    args:
      - --destination=gcr.io/$PROJECT_ID/$_IMG_NAME-prod:$SHORT_SHA
      - --cache=true
      - --cache-ttl=2h
      - --build-arg=FRONTEND_STAGE=prod
      - --build-arg=SHORT_SHA=$SHORT_SHA
```
Dockerfile
```
FROM node:10

ARG FRONTEND_STAGE=dev

ARG SHORT_SHA
ENV SHORT_SHA=$SHORT_SHA

WORKDIR /app
COPY package*.json ./

COPY id_rsa /root/.ssh/id_rsa
COPY known_hosts /root/.ssh/known_hosts
RUN eval $(ssh-agent) && \
    ssh-add /root/.ssh/id_rsa
RUN ls /root/.ssh

RUN npm i

COPY . .

EXPOSE 8080

RUN npm run build:$FRONTEND_STAGE

CMD [ "npm", "run", "dist" ]
```

Docker info
```
USER@cloudshell:~ (PROJ)$ docker version
Client: Docker Engine - Community
 Version:           19.03.5
 API version:       1.39 (downgraded from 1.40)
 Go version:        go1.12.12
 Git commit:        633a0ea838
 Built:             Wed Nov 13 07:25:58 2019
 OS/Arch:           linux/amd64
 Experimental:      false
Server: Docker Engine - Community
 Engine:
  Version:          18.09.6
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.8
  Git commit:       481bc77
  Built:            Sat May  4 01:59:36 2019
  OS/Arch:          linux/amd64
  Experimental:     false
```

Another user got an error from docker: `failed to register layer: ApplyLayer duplicates of file paths not supported stdout: {"layerSize":21436138}
 stderr:`

when extracting the same layer. Trying to get more info there.

**Additional Information**
Also reported in https://github.com/GoogleContainerTools/kaniko/issues/1024#issuecomment-581937390 I think

I wanted to open this issue in case others are seeing the same. I'll try to get kaniko debug builds and possibly docker debug logs too and add them here

 **Triage Notes for the Maintainers**
 

 
 | **Description** | **Yes/No** |
 |----------------|---------------|
 | Please check if this a new feature you are proposing        | <ul><li>- [ ] </li></ul>|
 | Please check if the build works in docker but not in kaniko | <ul><li>- [x] </li></ul>| 
 | Please check if this error is seen when you use `--cache` flag | <ul><li>- [x] </li></ul>|
 | Please check if your dockerfile is a multistage dockerfile | <ul><li>- [ ] </li></ul>|

**Comment by RemcodM** [comment]
I am experiencing this exact issue. The error message differs though. But it seems indeed that the layers produced by kaniko seem corrupt somehow. Layers from base images seem to be uploaded correct to my own registry, but image layers build by kaniko fail to extract or sometimes they even fail to verify (so verification of the layer fails).

I do not experience this problem when building with 0.16.0 or when building the Dockerfiles in Docker natively.

**Comment by afirth** [comment]
I cannot reproduce this with a minimal cloudbuild of the same dockerfile. However, in my investigation I found a race in our cloudbuild.yaml (running `npm test` in a separate step in parallel, polluting workspace and then deleting those files while kaniko was running). I'm going to leave this open for now as others are reporting and thumbsupping, so I think behavior has changed.

**Comment by RemcodM** [comment]
I am not running any background or parallel tasks while building in GitLab CI. The Dockerfile does also not contain any such behaviour... Therefore I would say the issue still exists. 

I have no time to test it right now, but when needed I can do some builds with my Dockerfiles under 0.17.1 later. Hopefully, I will be able to reproduce the behaviour and post the related information here.

**Comment by afirth** [comment]
> I am not running any background or parallel tasks while building in GitLab CI. The Dockerfile does also not contain any such behaviour... Therefore I would say the issue still exists.
> 
> I have no time to test it right now, but when needed I can do some builds with my Dockerfiles under 0.17.1 later. Hopefully, I will be able to reproduce the behaviour and post the related information here.

would be great. We saw some interesting things in the docker daemon log (which I was hoping to put in debug mode with a repro image).

**Comment by hstenzel** [comment]
Agree. We're having this same issue. Thanks @afirth for posting the workaround.


Please check if this a new feature you are proposing |  
-- | --
Please check if the build works in docker but not in kaniko |  true
Please check if this error is seen when you use --cache flag |  true
Please check if your dockerfile is a multistage dockerfile | true

**Comment by stevehipwell** [comment]
We've got the exact same issue running v0.17.1, but not on v0.16.0.

**Comment by cvgw** [comment]
I suspect this is related to #1039

**Comment by alejandrox1** [comment]
+1 to this
I was getting an error about "/lib/systemd /usr/lib/systemd" not being found (im using the latest version to build).

**Comment by cvgw** [comment]
This should be fixed. Can you try tag `a1af057f997316bfb1c4d2d82719d78481a02a79` or `debug-a1af057f997316bfb1c4d2d82719d78481a02a79`

**Comment by afirth** [comment]
> This should be fixed. Can you try tag `a1af057f997316bfb1c4d2d82719d78481a02a79` or `debug-a1af057f997316bfb1c4d2d82719d78481a02a79`

I can't repro those original broken builds since removing our npm race, sorry

**Comment by tejal29** [comment]
Thanks @afirth for explaining the issue. I verified an npm install and i no longer see this issue. Closing this now.
